CMakeFiles/xintface.dir/Cintface.c.o: relocation R_X86_64_32S against symbol `stdout@@GLIBC_2.2.5' can not be used when making a PIE object

See attached [log](https://github.com/Reference-ScaLAPACK/scalapack/files/8206502/configure.log).

```
                  Error configuring SCALAPACK with cmake Could not execute "['/nfs/gce/software/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/cmake-3.20.5-yjp2hz6/bin/cmake .. -DCMAKE_INSTALL_PREFIX=/home/svcpetsc/petsc-hash-pkgs/950b03 -DCMAKE_INSTALL_NAME_DIR:STRING="/home/svcpetsc/petsc-hash-pkgs/950b03/lib" -DCMAKE_INSTALL_LIBDIR:STRING="lib" -DCMAKE_VERBOSE_MAKEFILE=1 -DCMAKE_BUILD_TYPE=Debug -DCMAKE_C_COMPILER="/home/svcpetsc/petsc-hash-pkgs/950b03/bin/mpicc" -DMPI_C_COMPILER="/home/svcpetsc/petsc-hash-pkgs/950b03/bin/mpicc" -DCMAKE_AR=/usr/bin/ar -DCMAKE_RANLIB=/usr/bin/ranlib -DCMAKE_C_FLAGS:STRING="-fPIC -fstack-protector -Qunused-arguments -g -O -fopenmp" -DCMAKE_C_FLAGS_DEBUG:STRING="-fPIC -fstack-protector -Qunused-arguments -g -O -fopenmp" -DCMAKE_C_FLAGS_RELEASE:STRING="-fPIC -fstack-protector -Qunused-arguments -g -O -fopenmp" -DCMAKE_CXX_COMPILER="/home/svcpetsc/petsc-hash-pkgs/950b03/bin/mpicxx" -DMPI_CXX_COMPILER="/home/svcpetsc/petsc-hash-pkgs/950b03/bin/mpicxx" -DCMAKE_CXX_FLAGS:STRING="-fstack-protector -g -O -fopenmp -fPIC -fopenmp" -DCMAKE_CXX_FLAGS_DEBUG:STRING="-fstack-protector -g -O -fopenmp -fPIC -fopenmp" -DCMAKE_CXX_FLAGS_RELEASE:STRING="-fstack-protector -g -O -fopenmp -fPIC -fopenmp" -DCMAKE_Fortran_COMPILER="/home/svcpetsc/petsc-hash-pkgs/950b03/bin/mpif90" -DMPI_Fortran_COMPILER="/home/svcpetsc/petsc-hash-pkgs/950b03/bin/mpif90" -DCMAKE_Fortran_FLAGS:STRING="-fPIC -ffree-line-length-0 -g -O -fopenmp" -DCMAKE_Fortran_FLAGS_DEBUG:STRING="-fPIC -ffree-line-length-0 -g -O -fopenmp" -DCMAKE_Fortran_FLAGS_RELEASE:STRING="-fPIC -ffree-line-length-0 -g -O -fopenmp" -DCMAKE_EXE_LINKER_FLAGS:STRING=" -fopenmp -fopenmp" -DBUILD_SHARED_LIBS:BOOL=ON -DLAPACK_LIBRARIES="-Wl,-rpath,/home/svcpetsc/petsc-hash-pkgs/950b03/lib -L/home/svcpetsc/petsc-hash-pkgs/950b03/lib -lflapack -Wl,-rpath,/home/svcpetsc/petsc-hash-pkgs/950b03/lib -L/home/svcpetsc/petsc-hash-pkgs/950b03/lib -lfblas -lm -lstdc++ -ldl -Wl,-rpath,/home/svcpetsc/petsc-hash-pkgs/950b03/lib -L/home/svcpetsc/petsc-hash-pkgs/950b03/lib -lmpi_usempif08 -lmpi_usempi_ignore_tkr -lmpi_mpifh -lmpi -lgfortran -lm -Wl,-rpath,/usr/lib/gcc/x86_64-linux-gnu/7 -L/usr/lib/gcc/x86_64-linux-gnu/7 -Wl,-rpath,/home/svcpetsc/petsc-hash-pkgs/950b03/lib -lgfortran -lm -lgcc_s -lquadmath -lpthread" --verbose=1']":
CMake Error: Unknown argument --verbose=1
CMake Error: Run 'cmake --help' for all supported options.
```

--- a/BLACS/INSTALL/CMakeLists.txt
+++ b/BLACS/INSTALL/CMakeLists.txt
@@ -2,3 +2,4 @@ cmake_minimum_required(VERSION 2.8)
 project(INSTALL C Fortran)
 
 add_executable(xintface Fintface.f Cintface.c)
+set_property(TARGET xintface PROPERTY POSITION_INDEPENDENT_CODE TRUE)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -102,6 +102,21 @@ set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${SCALAPACK_BINARY_DIR}/lib)
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${SCALAPACK_BINARY_DIR}/lib)
 
 # --------------------------------------------------
+# By default static library
+function (invertBoolean varName varValue)
+  if (${varValue})
+    set(${varName} false PARENT_SCOPE)
+  else ()
+    set(${varName} true PARENT_SCOPE)
+  endif ()
+endfunction ()
+option(BUILD_SHARED_LIBS "Build shared libraries" OFF )
+invertBoolean("BUILD_STATIC_LIBS" ${BUILD_SHARED_LIBS})
+if ((${BUILD_SHARED_LIBS} EQUAL ON) AND NOT CMAKE_POSITION_INDEPENDENT_CODE)
+  set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+endif ()
+
+# --------------------------------------------------
 # Check for any necessary platform specific compiler flags
 include( CheckBLACSCompilerFlags )
 CheckBLACSCompilerFlags()
@@ -175,11 +190,6 @@ message(STATUS "BLAS library: ${BLAS_LIBRARIES}")
 message(STATUS "LAPACK library: ${LAPACK_LIBRARIES}")
 
 MESSAGE(STATUS "=========")
-
-# --------------------------------------------------
-# By default static library
-OPTION(BUILD_SHARED_LIBS "Build shared libraries" OFF )
-OPTION(BUILD_STATIC_LIBS "Build static libraries" ON )
 
 # --------------------------------------------------
 # Subdirectories that need to be processed

